"Unknown Python implementation" causing all PRs to fail CI

```
+ meson install --destdir /tmp/pip-install-pghsgssg/scipy_b92ebf441ebc4717b7fc4cfbd2105fe7/.mesonpy-x8l7n4ir/install
      Traceback (most recent call last):
        File "/opt/hostedtoolcache/PyPy/3.8.13/x64/lib/pypy3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 156, in prepare_metadata_for_build_wheel
          hook = backend.prepare_metadata_for_build_wheel
      AttributeError: module 'mesonpy' has no attribute 'prepare_metadata_for_build_wheel'
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "/tmp/pip-build-env-ptqp77gk/overlay/lib/pypy3.8/site-packages/mesonpy/__init__.py", line 726, in _calculate_file_abi_tag_heuristic_posix
          return mesonpy._tags.StableABITag(tag)
        File "/tmp/pip-build-env-ptqp77gk/overlay/lib/pypy3.8/site-packages/mesonpy/_tags.py", line 39, in __init__
          raise ValueError(f'Invalid PEP 3149 stable ABI tag, expecting pattern `{self._REGEX.pattern}`')
      ValueError: Invalid PEP 3149 stable ABI tag, expecting pattern `^abi(?P<abi_number>[0-9]+)$`
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "/opt/hostedtoolcache/PyPy/3.8.13/x64/lib/pypy3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/opt/hostedtoolcache/PyPy/3.8.13/x64/lib/pypy3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/opt/hostedtoolcache/PyPy/3.8.13/x64/lib/pypy3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 160, in prepare_metadata_for_build_wheel
          whl_basename = backend.build_wheel(metadata_directory, config_settings)
        File "/tmp/pip-build-env-ptqp77gk/overlay/lib/pypy3.8/site-packages/mesonpy/__init__.py", line 941, in build_wheel
          return project.wheel(out).name
        File "/tmp/pip-build-env-ptqp77gk/overlay/lib/pypy3.8/site-packages/mesonpy/__init__.py", line 889, in wheel
          wheel = _WheelBuilder(self).build(self._install_plan, self._copy_files, self._build_dir)
        File "/tmp/pip-build-env-ptqp77gk/overlay/lib/pypy3.8/site-packages/mesonpy/__init__.py", line 356, in build
          wheel_file = pathlib.Path(directory, f'{self.name}.whl')
        File "/tmp/pip-build-env-ptqp77gk/overlay/lib/pypy3.8/site-packages/mesonpy/__init__.py", line 165, in name
          python_tag=self._project.python_tag,
        File "/tmp/pip-build-env-ptqp77gk/overlay/lib/pypy3.8/site-packages/mesonpy/__init__.py", line 657, in python_tag
          selected_tag = self._select_abi_tag()
        File "/tmp/pip-build-env-ptqp77gk/overlay/lib/pypy3.8/site-packages/mesonpy/__init__.py", line 793, in _select_abi_tag
          tags = self._files_by_tag()
        File "/tmp/pip-build-env-ptqp77gk/overlay/lib/pypy3.8/site-packages/mesonpy/__init__.py", line 773, in _files_by_tag
          tag = self._calculate_file_abi_tag_heuristic(file)
        File "/tmp/pip-build-env-ptqp77gk/overlay/lib/pypy3.8/site-packages/mesonpy/__init__.py", line 735, in _calculate_file_abi_tag_heuristic
          return self._calculate_file_abi_tag_heuristic_posix(filename)
        File "/tmp/pip-build-env-ptqp77gk/overlay/lib/pypy3.8/site-packages/mesonpy/__init__.py", line 728, in _calculate_file_abi_tag_heuristic_posix
          return mesonpy._tags.LinuxInterpreterTag(tag)
        File "/tmp/pip-build-env-ptqp77gk/overlay/lib/pypy3.8/site-packages/mesonpy/_tags.py", line 79, in __init__
          f'Unknown Python implementation: {self.implementation}. '
      NotImplementedError: Unknown Python implementation: pypy38. Please report this to https://github.com/FFY00/mesonpy/issues and include information about the Python distribution you are using.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/runtests.yml
+++ b/.github/workflows/runtests.yml
@@ -197,9 +197,11 @@ jobs:
       - if: ${{ contains(matrix.python-version, '3.11') }}
         run: pip install git+https://github.com/matplotlib/matplotlib@main
 
-      # SciPy 1.9.0 fails to compile under PyPy
+      # SciPy 1.9.0 and 1.9.1 fail to compile under PyPy:
+      #
+      # https://github.com/FFY00/meson-python/issues/121
       - if: ${{ matrix.python-version == 'pypy-3.8' }}
-        run: pip install scipy!=1.9.0
+        run: pip install 'scipy<1.9.0'
 
       # dependencies to install in all Python versions:
       - run: pip install mpmath numpy numexpr matplotlib ipython cython scipy \

